### PR TITLE
fix(poll-command): fixed poll command to support merged state root

### PR DESCRIPTION
**Pull request by crisgarner**
# Description

Fixed poll command to support merged state root

## Confirmation

- [x] I have read and understand MACI's [contributor guidelines](https://maci.pse.dev/docs/contributing) and [code of conduct](https://maci.pse.dev/docs/contributing/code-of-conduct).
- [x] I have read and understand MACI's [GitHub processes](https://github.com/privacy-scaling-explorations/maci/discussions/847).
- [x] I have read and understand MACI's [testing guide](https://maci.pse.dev/docs/testing).

### CODE REVIEW

File: cli/ts/commands/poll.ts [hunk 1]
```diff
@@ -55,7 +55,7 @@ export const getPoll = async ({
         `Deploy time: ${new Date(Number(deployTime) * 1000).toString()}`,
         `End time: ${new Date(Number(deployTime + duration) * 1000).toString()}`,
         `Number of signups ${numSignups}`,
-        `State Aq merged: ${isStateAqMerged}`,
+        `State Aq merged: ${mergedStateRoot}`,
```
Comment (ctrlc03) [inline cli/ts/commands/poll.ts:58]: could you please change the text to `State tree merged`? to indicate is not an Aq anymore

File: cli/ts/commands/poll.ts [hunk 2]
```diff
@@ -40,12 +40,12 @@ export const getPoll = async ({
...
 
-  const [[deployTime, duration], isStateAqMerged] = await Promise.all([
+  const [[deployTime, duration], mergedStateRoot] = await Promise.all([
     pollContract.getDeployTimeAndDuration(),
-    pollContract.stateMerged(),
+    pollContract.mergedStateRoot(),
   ]);
 
-  const numSignups = await (isStateAqMerged ? pollContract.numSignups() : maciContract.numSignUps());
+  const numSignups = await (mergedStateRoot ? pollContract.numSignups() : maciContract.numSignUps());
```
Comment (ctrlc03) [inline cli/ts/commands/poll.ts:48]: does this work as expected given the default value for when it's not set it's 0?

Comment (crisgarner) [reply cli/ts/commands/poll.ts:48]: > does this work as expected given the default value for when it's not set it's 0?

Ran the test suit and didn't revert, could be that we aren't testing that, we can be sure by adding a check that if it's not equal to 0

**Review by ctrlc03 — CHANGES_REQUESTED** [review]
Thanks, left two comments :)

File: cli/ts/utils/interfaces.ts [hunk 3]
```diff
@@ -950,7 +950,7 @@ export interface IGetPollData {
   /**
    * Whether the MACI contract's stateAq has been merged by this contract
```
Comment (kittybest) [inline cli/ts/utils/interfaces.ts:951]: then the description here could also be modified to `state root has been merged`

**Review by kittybest — COMMENTED** [review]
thanks! just a comment

File: cli/ts/utils/interfaces.ts [hunk 4]
```diff
@@ -950,7 +950,7 @@ export interface IGetPollData {
...
    */
-  isStateAqMerged: boolean;
+  mergedStateRoot: bigint;
```
Comment (ctrlc03) [inline cli/ts/utils/interfaces.ts:953]: I think we should keep this as a boolean and return whether it's 0 or not

**Review by ctrlc03 — APPROVED** [review]
LGTM thanks

**Pull request merged**